sematic/ui/packages/main/src/pipelines/RunMetricsPanel.tsx

```diff
@@ -0,0 +1,103 @@
+import { Timeline } from "@mui/icons-material";
+import { Box, Typography, useTheme } from "@mui/material";
+import { useEffect, useMemo, useState } from "react";
+import { Link } from "react-router-dom";
+import Loading from "src/components/Loading";
+import { TimeseriesMetric } from "src/components/Metrics";
+import { METRIC_SCOPES } from "@sematic/common/src/constants";
+import { MetricsFilter, useListMetrics } from "@sematic/common/src/hooks/metricsHooks";
+import { usePipelinePanelsContext } from "src/hooks/pipelineHooks";
+import { getChartColor } from "src/utils";
+import {
+    MetricPoint,
+    runIsInTerminalState,
+} from "@sematic/common/lib/src/Models";
+import { metricsSocket } from "src/sockets";
+
+export default function RunMetricsPanel() {
+    const theme = useTheme();
+
+    const { selectedRun } = usePipelinePanelsContext();
+
+    let run = selectedRun!;
+
+    const [latestBroadcastEvent, setLatestBroadcastEvent] = useState<
+    MetricPoint[] | undefined
+    >();
+
+    useEffect(() => {
+        if (runIsInTerminalState(run)) return;
+        metricsSocket.removeAllListeners("update");
```

Comment:
Now I think this is a bad idea because it makes socket usage exclusive to a single component. ( I regret that I used this pattern elsewhere, but most likely I copied it too)

The right way should be to unsubscribe when the component is unmounted (using useEffect). 


```
const eventListener = useCallback(...) 

useEffect({
   metricsSocket.on("update", eventListener);
   return () => {
        metricsSocket.off("update", eventListener);
   };
}, []);
```